contrib/golang/filters/http/source/golang_filter.cc

```diff
@@ -344,13 +345,17 @@ bool Filter::doTrailer(ProcessorState& state, Http::HeaderMap& trailers) {
     // do data first
     if (!state.isBufferDataEmpty()) {
       done = doDataGo(state, state.getBufferData(), false);
-    }
-    // NP: can not use done as condition here, since done will be false
-    // maybe we can remove the done variable totally? by using state_ only?
-    // continue trailers
-    if (state.state() == FilterState::WaitingTrailer) {
+      // NP: can not use done as condition here, since done will be false
+      // maybe we can remove the done variable totally? by using state_ only?
+      // continue trailers
+      if (state.state() == FilterState::WaitingTrailer) {
+        done = doTrailerGo(state, trailers);
+      }
+    } else {
```

Comment:
I'm not sure about these changes, could you please explain them?

Reply:
just like what i comment above, `doTrailerGo` should also be called when `isBufferDataEmpty` is true, otherwize the filter chain will be stopped.

Reply:
oh, yes, I see, didn't see the comment before I comment, thanks.